clusterctl/examples/openstack/provider-components.yaml.template

```diff
@@ -0,0 +1,292 @@
+data:
+  config_config.yaml: $OPENSTACK_CLOUD_CONFIG
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: machine-setup
+data:
+  machine_setup_configs.yaml: |-
+    items:
+    - machineParams:
+      - os: ubuntu-1604-lts
+        roles:
+        - Master
+        versions:
+          kubelet: 1.9.4
+          controlPlane: 1.9.4
+          containerRuntime:
+            name: docker
+            version: 1.12.0
+      metadata:
+        startupScript: |
+          set -e
+          set -x
+          (
+          ARCH=amd64
+          curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+          touch /etc/apt/sources.list.d/kubernetes.list
+          sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
+          apt-get update -y
+          apt-get install -y \
+              prips
+
+          function install_configure_docker () {
+              # prevent docker from auto-starting
+              echo "exit 101" > /usr/sbin/policy-rc.d
+              chmod +x /usr/sbin/policy-rc.d
+              trap "rm /usr/sbin/policy-rc.d" RETURN
+              apt-get install -y docker.io
+              echo 'DOCKER_OPTS="--iptables=false --ip-masq=false"' > /etc/default/docker
+              systemctl daemon-reload
+              systemctl enable docker
+              systemctl start docker
+          }
+          install_configure_docker
+
+          curl -sSL https://dl.k8s.io/release/${VERSION}/bin/linux/${ARCH}/kubeadm > /usr/bin/kubeadm.dl
+          chmod a+rx /usr/bin/kubeadm.dl
+          # kubeadm uses 10th IP as DNS server
+          CLUSTER_DNS_SERVER=$(prips ${SERVICE_CIDR} | head -n 11 | tail -n 1)
+          # Our Debian packages have versions like "1.8.0-00" or "1.8.0-01". Do a prefix
+          # search based on our SemVer to find the right (newest) package version.
+          function getversion() {
+              name=$1
+              prefix=$2
+              version=$(apt-cache madison $name | awk '{ print $3 }' | grep ^$prefix | head -n1)
+              if [[ -z "$version" ]]; then
+                  echo Can\'t find package $name with prefix $prefix
+                  exit 1
+              fi
+              echo $version
+          }
+          KUBELET=$(getversion kubelet ${KUBELET_VERSION}-)
+          KUBEADM=$(getversion kubeadm ${KUBELET_VERSION}-)
+          apt-get install -y \
+              kubelet=${KUBELET} \
+              kubeadm=${KUBEADM}
+          mv /usr/bin/kubeadm.dl /usr/bin/kubeadm
+          chmod a+rx /usr/bin/kubeadm
+
+          cat > /etc/systemd/system/kubelet.service.d/20-kubenet.conf <<EOF
+          [Service]
+          Environment="KUBELET_DNS_ARGS=--cluster-dns=${CLUSTER_DNS_SERVER} --cluster-domain=${CLUSTER_DNS_DOMAIN}"
+          EOF
+          systemctl daemon-reload
+          systemctl restart kubelet.service
+
+          # Set up kubeadm config file to pass parameters to kubeadm init.
```

Comment:
looks like the k8s we install does not use the cloud provider, we can do that later. no rush